Add OffScreenCanvas context lost/restored events (#33159)
mdn/content: files/en-us/web/api/offscreencanvas/contextrestored_event/index.md

---
title: "OffscreenCanvas: contextrestored event"
short-title: contextrestored
slug: Web/API/OffscreenCanvas/contextrestored_event
page-type: web-api-event
status:
  - experimental
browser-compat: api.OffscreenCanvas.contextrestored_event
---

> {{APIRef}}{{SeeCompatTable}}

Reviewer: ```suggestion
{{APIRef("Canvas API")}}{{SeeCompatTable}}
```

> The **`contextrestored`** event of the [Canvas API](/en-US/docs/Web/API/Canvas_API) is fired if the user agent restores the backing storage for a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D).

Reviewer: ```suggestion
The **`contextrestored`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the user agent restores the backing storage for a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D).
```

> The **`contextrestored`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the user agent restores the backing storage for a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D).

Author: ```suggestion
The **`contextrestored`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the browser restores the underlying system providing resources for a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D).
```

> The **`contextrestored`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the browser restores the underlying system, such as a GPU, that provides resources for a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context.

Reviewer 2: ```suggestion
The **`contextrestored`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the browser restores a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context, for example, after a GPU process has crashed.
```

Author: Here I'm going to link to the "context lost" event, as that gives examples of why things were lost. At this point you don't need to know why it was lost and how it was restored because the next line says what you need to _do_.

```suggestion
The **`contextrestored`** event of the {{domxref("OffscreenCanvas")}} interface is fired if the browser restores a [`OffscreenCanvasRenderingContext2D`](/en-US/docs/Web/API/OffscreenCanvasRenderingContext2D) context that was [previously lost](/en-US/docs/Web/API/OffscreenCanvas/contextlost_event).
```

You can redraw, re-retrieve resources, and reinitialize the state of your context after receiving this event.

## Syntax

Use the event name in methods like {{domxref("EventTarget.addEventListener", "addEventListener()")}}, or set an event handler property.

```js
addEventListener("contextrestored", (event) => {});

oncontextrestored = (event) => {};
```

## Event type

A generic {{domxref("Event")}}.

> ## Example

Reviewer: ```suggestion
## Examples
```

Author: Yup. I copied this, so presumably the original is wrong too :-(
